src/main/java/net/neoforged/neoforge/common/conditions/IConditionBuilder.java

```diff
@@ -33,16 +35,36 @@ default ICondition or(ICondition... values) {
         return new OrCondition(List.of(values));
     }
 
+    default ICondition itemExists(ResourceLocation itemName) {
+        return new ItemExistsCondition(itemName);
+    }
+
     default ICondition itemExists(String namespace, String path) {
-        return new ItemExistsCondition(namespace, path);
+        return new ItemExistsCondition(ResourceLocation.fromNamespaceAndPath(namespace, path));
+    }
+
+    default ICondition itemExists(String itemName) {
+        return new ItemExistsCondition(ResourceLocation.parse(itemName));
     }
 
     default ICondition modLoaded(String modid) {
         return new ModLoadedCondition(modid);
     }
 
     default ICondition tagEmpty(TagKey<Item> tag) {
-        return new TagEmptyCondition(tag.location());
+        return new TagEmptyCondition(tag);
+    }
+
+    default ICondition tagEmpty(ResourceLocation itemTag) {
+        return tagEmpty(ItemTags.create(itemTag));
+    }
+
+    default ICondition tagEmpty(String namespace, String tagPath) {
+        return tagEmpty(ResourceLocation.fromNamespaceAndPath(namespace, tagPath));
+    }
+
+    default ICondition tagEmpty(String tagPath) {
```

Comment:
`path` is misleading since it also includes the namespace. I'd probably remove the `String` methods, that's an outdated pattern. :wink: